### PR TITLE
blog: gateway vs ingress API

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

New blog post about Kubernetes Gateway and Ingress APIs.

### CODE REVIEW

File: content/posts/gateway-vs-ingress-api.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+Annotations are specific to your choice of an Ingress controller. These "proprietary" extensions limited the scope of portability intended initially with the Ingress API.
+
+## Custom CRDs > Ingress API
+
+Being stuck with annotations also sacrifice the usability of the Ingress controllers.
+
+Controllers therefore solved the limitations of the Ingress API by creating their [own custom resources](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/). The example below shows configuring Ingress to route traffic between two versions of an application using APISIX's custom resource:
+
+```yaml {title="apisix-ingress-manifest.yaml"}
+apiVersion: apisix.apache.org/v2beta3
```
Comment (AlinsRan) [inline content/posts/gateway-vs-ingress-api.md]: https://apisix.apache.org/docs/ingress-controller/concepts/apisix_route/

In APISIX Ingress v1.5 version, we recommend use `apisix.apache.org/v2`. The v2beta3 has been marked as deprecated.

File: content/posts/gateway-vs-ingress-api.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+          weight: 90
+        - serviceName: bare-minimum-api-v2
+          servicePort: 8081
+          weight: 10
+```
+
+Traffic split with Gateway API (see [Canary traffic rollout](https://gateway-api.sigs.k8s.io/guides/traffic-splitting/#canary-traffic-rollout)):
+
+```yaml {title="kubernetes-gateway-manifest.yaml"}
+apiVersion: gateway.networking.k8s.io/v1beta1
```
Comment (AlinsRan) [inline content/posts/gateway-vs-ingress-api.md]: https://apisix.apache.org/docs/ingress-controller/aeps/gateway-api/

In APISIX Ingres, only `v1alpha2` is supported, and `v1beta1` does not support.

File: content/posts/gateway-vs-ingress-api.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+{{< /blockquote >}}
+
+It takes inspiration from the custom CRDs of different Ingress controllers mentioned earlier.
+
+The Gateway API adds [many features](https://gateway-api.sigs.k8s.io/#gateway-api-concepts) "on top" of the Ingress API's capabilities. This includes HTTP header-based matching, weighted traffic splitting, and other features that require custom proprietary annotations with the Ingress API.
+
+Traffic split with APISIX Ingress resource (see [ApisixRoute/v2beta3 reference](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/)):
+
+```yaml {title="apisix-ingress-manifest.yaml"}
+apiVersion: apisix.apache.org/v2beta3
```
Comment (AlinsRan) [inline content/posts/gateway-vs-ingress-api.md]: Ditto.

**Comment by pottekkat** [comment]
Thanks @AlinsRan I have made the suggested changes.

File: content/posts/gateway-vs-ingress-api.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        direction TB
+        ic
+        p
+    end
+{{< /mermaid >}}
+
+
+You can create an [Ingress resource](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource) to configure APISIX or any other Ingress implementations.
+
+The example below shows how you can route traffic between two versions of an application with APISIX Ingress:
```
Comment (tao12345666333) [inline content/posts/gateway-vs-ingress-api.md:76]: This requires the app to be able to distinguish between different versions via /v1 and /v2

File: content/posts/gateway-vs-ingress-api.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+This creates an Ingress resource that configures APISIX to route any requests with the `/app` prefix to the backend with the prefix removed. For example, a request to `/app/version` will be forwarded to `/version`.
+
+Annotations are specific to your choice of an Ingress controller. These "proprietary" extensions limited the scope of portability intended initially with the Ingress API.
+
+## Custom CRDs > Ingress API
+
+Being stuck with annotations also sacrifice the usability of the Ingress controllers.
+
+Controllers therefore solved the limitations of the Ingress API by creating their [own custom resources](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/). The example below shows configuring Ingress to route traffic between two versions of an application using APISIX's custom resource:
```
Comment (tao12345666333) [inline content/posts/gateway-vs-ingress-api.md]: Change this to v2

File: content/posts/gateway-vs-ingress-api.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+{{< blockquote author="gateway-api.sigs.k8s.io" link="[Link](https://gateway-api.sigs.k8s.io/)" title="What is the Gateway API?" >}}
+(Gateway API) aim to evolve Kubernetes service networking through expressive, extensible, and role-oriented interfaces ...
+{{< /blockquote >}}
+
+It takes inspiration from the custom CRDs of different Ingress controllers mentioned earlier.
+
+The Gateway API adds [many features](https://gateway-api.sigs.k8s.io/#gateway-api-concepts) "on top" of the Ingress API's capabilities. This includes HTTP header-based matching, weighted traffic splitting, and other features that require custom proprietary annotations with the Ingress API.
+
+Traffic split with APISIX Ingress resource (see [ApisixRoute/v2beta3 reference](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/)):
```
Comment (tao12345666333) [inline content/posts/gateway-vs-ingress-api.md]: Ditto

File: content/posts/gateway-vs-ingress-api.md (hunk 4 above)
Comment (pottekkat) [reply content/posts/gateway-vs-ingress-api.md:76]: `/v1` will go to one service and `/v2` will go to the other.

**Pull request merged**